simtools/utils/names.py

```diff
@@ -257,322 +249,202 @@ def _validate_name(name, all_names):
         If name is not valid.
     """
 
-    if not _is_valid_name(name, all_names):
-        msg = f"Invalid name {name}"
-        raise ValueError(msg)
-    for main_name, list_of_names in all_names.items():
-        if name.lower() in list_of_names + [main_name.lower()]:
-            if name != main_name:
-                _logger.debug(f"Correcting name {name} -> {main_name}")
-            return main_name
-    return None
-
-
-def _is_valid_name(name, all_names):
-    """
-    Check if name is valid.
+    for key in all_names.keys():
+        if isinstance(all_names[key], list) and name.lower() in [
+            item.lower() for item in all_names[key]
+        ]:
+            return key
+        if name.lower() == key.lower():
+            return key
 
-    Parameters
-    ----------
-    name:  str
-        Name to validated.
-    all_names: dict
-        Dictionary with valid names.
-
-    Returns
-    -------
-    bool
-        True if name is valid. Otherwise, false.
-    """
-
-    if not isinstance(name, str):
-        return False
-    for main_name in all_names.keys():
-        if name.lower() in all_names[main_name] + [main_name.lower()]:
-            return True
-    return False
+    msg = f"Invalid name {name}"
+    raise ValueError(msg)
 
 
-def validate_telescope_model_name(name):
+def validate_telescope_name(name):
     """
-    Validate a telescope model name.
+    Validate telescope name (e.g., MSTN-design, MSTN-01).
 
     Parameters
     ----------
     name: str
-        Telescope model name.
+        Telescope name.
 
     Returns
     -------
     str
         Validated name.
     """
-
-    # e.g, MSTN or MSTN-01
     try:
-        return _validate_name(name, all_array_element_id_names)
-    except ValueError:
-        pass
-
-    # e.g., MST-FlashCam or MST-FlashCam-01
-    tel_class, tel_type, tel_id = split_telescope_model_name(name)
-    _telescope_name = tel_class
-    if tel_class != tel_type and len(tel_type) > 0:
-        _telescope_name += "-" + tel_type
-    if len(tel_id) > 0:
-        _telescope_name += "-" + tel_id
-    return _telescope_name
+        _tel_type, _tel_id = name.split("-")
+    except ValueError as exc:
```

Comment:
When would this return a ValueError?

Reply:
e.g. `_tel_type, _tel_id = "LSTN".split("-")` , as it has only one return parameter ("LSTN").